Core/src/org/sleuthkit/autopsy/centralrepository/eventlisteners/CaseEventListener.java

```diff
@@ -154,82 +206,90 @@ private ContentTagTask(CentralRepository db, PropertyChangeEvent evt) {
+            //   - If there are other notable tags, the comment should be updated to most recent tag
+
+            if (!isNotableTagName(tagDeletedEvent.getDeletedTagInfo().getName())) {
+                // If the tag that got removed isn't on the list of central repo tags, do nothing
+                return;
+            }
+
+            try {
+                // Get the remaining tags on the content object
                 long contentID = tagDeletedEvent.getDeletedTagInfo().getContentID();
+                Content content = Case.getCurrentCaseThrows().getSleuthkitCase().getContentById(contentID);
+                TagsManager tagsManager = Case.getCurrentCaseThrows().getServices().getTagsManager();
+                List<ContentTag> tags = tagsManager.getContentTagsByContent(content);
 
-                String tagName = tagDeletedEvent.getDeletedTagInfo().getName().getDisplayName();
-                if (!TagsManager.getNotableTagDisplayNames().contains(tagName)) {
-                    // If the tag that got removed isn't on the list of central repo tags, do nothing
+                AbstractFile af = getAbstractFile(content);
+                if (af == null)
                     return;
-                }
 
-                try {
-                    // Get the remaining tags on the content object
-                    Content content = Case.getCurrentCaseThrows().getSleuthkitCase().getContentById(contentID);
-                    TagsManager tagsManager = Case.getCurrentCaseThrows().getServices().getTagsManager();
-                    List<ContentTag> tags = tagsManager.getContentTagsByContent(content);
-
-                    if (tags.stream()
-                            .map(tag -> tag.getName().getDisplayName())
-                            .filter(TagsManager.getNotableTagDisplayNames()::contains)
-                            .collect(Collectors.toList())
-                            .isEmpty()) {
-
-                        // There are no more bad tags on the object
-                        if (content instanceof AbstractFile) {
-                            af = (AbstractFile) content;
-                            knownStatus = TskData.FileKnown.UNKNOWN;
-                            comment = "";
-                        } else {
-                            LOGGER.log(Level.WARNING, "Error updating non-file object");
-                            return;
-                        }
-                    } else {
-                        // There's still at least one bad tag, so leave the known status as is
-                        return;
-                    }
-                } catch (TskCoreException | NoCurrentCaseException ex) {
-                    LOGGER.log(Level.SEVERE, "Failed to find content", ex);
-                    return;
+                if (hasNotableTag(tags)) {
+                    // There's still at least one bad tag, so leave the known status but update the comment to most recent
```

Comment:
"Bad tags" is an outdated concept. Tags either confer notability on a file, or they do not. 

We no longer label files as "bad" in Autopsy. Files are either notable, known or unknown.  There are leftover references to "bad" all over the place, but we should not add new references nor keep old references to "badness" unless it happens to be a matter of public API.